pkg/aws/ec2/api/wrapper.go

```diff
@@ -477,8 +485,7 @@ func (e *ec2Wrapper) getClientUsingAssumedRole(instanceRegion string, roleARN st
 	// If the regional STS endpoint is different than the global STS endpoint then add the global sts endpoint
 	if regionalSTSEndpoint.URL != globalSTSEndpoint.URL {
 		globalProvider := &stscreds.AssumeRoleProvider{
-			Client: sts.New(userStsSession, aws.NewConfig().WithHTTPClient(client).
-				WithEndpoint(regionalSTSEndpoint.URL).WithMaxRetries(MaxRetries)),
```

Comment:
is this a bug fix? since the previous code here were using "regionalSTSEndpoint.URL" instead of globalSTSEndpoint. If so, why do we now want to enable globalSTS?

Reply:
I think this was a bug. I don't think we ever used global.

Reply:
Since we already compare `egionalSTSEndpoint.URL != globalSTSEndpoint.URL`, it makes sense to correct the endpoint to global one.

Reply:
After syncing offline, we decided to use the old way (both use regional endpoint) and will investigate if we should remove global endpoint.